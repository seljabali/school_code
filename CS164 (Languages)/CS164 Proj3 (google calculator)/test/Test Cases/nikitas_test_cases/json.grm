//JSON: JavaScript Object Notation
%%
O -> '{' FIELDS '}';
FIELDS -> ID ':' V FMORE;
FMORE -> ',' ID ':' V FMORE | _;
V -> O | /'[^']*'/ | ARR | /[1-9][0-9]*/; //objects, strings, arrays, numbers
ARR -> '[' ']' | '[' V AMORE ']';
AMORE -> ',' V AMORE | _;
ID -> /[a-zA-Z][^ :]*/;