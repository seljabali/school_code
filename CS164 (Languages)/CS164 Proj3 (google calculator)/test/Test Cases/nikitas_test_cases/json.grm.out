(Grammar
    (Declarations
    )
    (Rules
        (Rule O
            (production [ \{, FIELDS, \}, ]
            )
        )
        (Rule FIELDS
            (production [ ID, \:, V, FMORE, ]
            )
        )
        (Rule FMORE
            (production [ \,, ID, \:, V, FMORE, ]
            )
            (production [ (epsilon), ]
            )
        )
        (Rule V
            (production [ O, ]
            )
            (production [ '[^']*', ]
            )
            (production [ ARR, ]
            )
            (production [ [1-9][0-9]*, ]
            )
        )
        (Rule ARR
            (production [ \[, \], ]
            )
            (production [ \[, V, AMORE, \], ]
            )
        )
        (Rule AMORE
            (production [ \,, V, AMORE, ]
            )
            (production [ (epsilon), ]
            )
        )
        (Rule ID
            (production [ [a-zA-Z][^ :]*, ]
            )
        )
    )
    (StartSymbol O)
)
